compilers/concrete-optimizer/concrete-optimizer/src/dag/operator/operator.rs

```diff
@@ -104,6 +105,11 @@ pub enum Operator {
         input: OperatorIndex,
         out_precision: Precision,
     },
+    ChangePartition {
+        input: OperatorIndex,
+        src_partition: Option<ExternalPartition>,
+        dst_partition: Option<ExternalPartition>,
```

Comment:
Is it really possible to have `None, None` ? Or `Some(a), Some(a)` ? 

If so, maybe it would be worth having an assert when adding the op !

Reply:
Hmm actually it is not clear for me how this is supposed to work. Only one of them is expected to be something (though you support having both set) ? And the other (none) is supposed to signal the fact that it is free to have any partition ? 

My original point was checking that both were something different. Maybe we could have a quick chat, so that I am sure I got everything right ?

Reply:
I think the "change partition" is just an informative flag of the origin/destination partition for circuit inputs and results. We choosed to keep it more generic and compatible with eventual future use in the middle of a circuit.
Current used form are
ChangePartition(Some, None) on inputs
ChangePartition(None, Some) on results
E.g. future use
`ChangePartition(None, None)` could hint that if a conflict exists the partition change should happen here and not somewhere else.
ChangePartition(None, Some), could be use to express manually where a transition occurs, which is forced automatically today by the optimizer policy but better manual decision could be taken.
Also it's translated from MLIR, so it could be used there to have multi partition annoted code instead of relying on optimizer choosing partition and parameters.